File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& arr) {
        bool hold = false;
        int total_profit = 0;
        for(int i=1;i<arr.size();i++){
            if(arr[i]>arr[i-1] && !hold){
                total_profit-=arr[i-1];
                hold = true;
            }
            else if(arr[i]<arr[i-1] && hold){
                total_profit+=arr[i-1];
                hold = false;
            }
            if(hold && i==arr.size()-1)total_profit+=arr[i];
        }
        return total_profit;
    }
};

File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.java
class Solution {
    public int maxProfit(int[] arr) {
        int mx_profit = 0;
        int l =0, r = 1;
        while(r<arr.length){
            if(arr[r]<arr[r-1]){
                mx_profit+=arr[r-1] - arr[l];
                l=r;
            }
            r++;
        }
        if(l!=r-1)mx_profit+=arr[r-1]-arr[l];
        // mx_profit = Math.max(mx_profit, arr[arr.length - 1] - arr[0]);
        return mx_profit;
    }
}